R/reporter.R

```diff
@@ -18,6 +18,24 @@ Reporter <- R6::R6Class("Reporter",
     end_context =    function(context) {},
     end_reporter =   function() {},
 
+    out = NULL,
+
+    initialize = function() {
+      self$out <- stdout()
+    },
+
+    cat = function(..., file = NULL, sep = " ", fill = FALSE, labels = NULL,
+                   append = NULL) {
+      if (!is.null(file)) {
+        warning("file ignored", call. = FALSE)
+      }
+      if (!is.null(append)) {
+        warning("append ignored", call. = FALSE)
+      }
+
+      cat(..., file = self$out, sep = sep, fill = fill, labels = labels)
```

Comment:
Rather than having our own clone of `cat()`, I'd rather have something that's tweaked to make like a bit easier:

```R
cat_line <- function(...) {
  cat(..., "\n", sep = "")
}
```

Not sure what to call the version that doesn't add a nl